corehq/apps/domain/views/releases.py

```diff
@@ -49,8 +50,12 @@ def page_context(self):
 
     def post(self, request, *args, **kwargs):
         if self.manage_releases_form.is_valid():
-            self.manage_releases_form.save()
-            return redirect(self.urlname, self.domain)
+            success, error_message = self.manage_releases_form.save()
+            if success:
+                return redirect(self.urlname, self.domain)
+            else:
+                messages.error(request, error_message)
+                return self.get(request, *args, **kwargs)
```

Comment:
nit: this feels like a messy interface - could you move the validation check to form.clean?  It's a user error, so it'd make sense to raise there.

Reply:
I actually made an effort to move the validation message to model in this [commit](https://github.com/dimagi/commcare-hq/pull/23680/commits/e61920e2328af091667f47d78408af01af3546ea) so that its not duplicated and hence its not a part of form clean.

Reply:
offlined.